odoo/tools/cache.py

```diff
@@ -191,3 +229,53 @@ def get_cache_key_counter(bound_method: Callable, *args, **kwargs):
     cache, key0, counter = ormcache_instance.lru(model)
     key = key0 + ormcache_instance.key(model, *args, **kwargs)
     return cache, key, counter
+
+
+def get_cache_size(
+        obj,
+        *,
+        cache_info: str = '',
+        seen_ids: set[int] | None = None,
+        class_slots: dict[type, Iterable[str]] | None = None
+    ) -> int:
+    """ A non-thread-safe recursive object size estimator """
```

Comment:
What about workers=0?
What if python run without GIL?

Reply:
You may just not get the exact size if objects are modified while the function is running; you may get more. However, objects in cache should be immutable so this is not an issue in reality.

Reply:
That's why I leave the message and expect it should only be used by signal handler, which always holds the GIL.